torch/csrc/jit/passes/remove_inplace_ops.cpp

```diff
@@ -38,6 +38,22 @@ void ImplicitCastForBinaryInplaceOps(Block* b) {
       if (originalInputs.at(0) == originalInputs.at(1)) {
         continue;
       }
+
+      // Introduce warning for special case where shape values are obtained from
+      // tensor.shape In eager mode these values are python int, so there are
+      // treated as different values. However in tracing both are traced as
+      // tensors, and then they share same memory. This causes a mismatch in
+      // final outputs
+      auto shape_node = originalInputs.at(0)->node();
+      if ((shape_node->kind() == prim::NumToTensor) &&
+          (shape_node->inputs().at(0)->node()->kind() == aten::size)) {
+        std::cerr
+            << "In tracing mode, shape values obtained from tensor.shape are traced as tensors, "
```

Comment:
I suggest we keep this very short here and just link to a longer explanation in onnx.html that includes example code.

Reply:
Would this go in the Frequently asked questions section?

Reply:
Would this go in the FAQ section?

Reply:
"Avoiding Pitfalls" or "Limitations" seem better to me. I can't imagine someone actually asking about this unless they see this warning, so it won't be a frequently asked question.